examples/http/client.cpp

```diff
@@ -0,0 +1,125 @@
+    opt_group{custom_options_, "tls"} //
+      .add<bool>("enable", "enables encryption via TLS")
+      .add<std::string>("ca-file", "CA file for trusted servers");
+  }
+
+  caf::settings dump_content() const override {
+    auto result = actor_system_config::dump_content();
+    caf::put_missing(result, "method", default_method);
+    caf::put_missing(result, "payload", "");
+    return result;
+  }
+};
+
+auto defatul_port(std::string_view scheme) {
+  if (scheme == "http")
+    return 80;
+  else if (scheme == "https")
+    return 443;
+  else
+    return 0;
+}
+
+int caf_main(caf::actor_system& sys, const config& cfg) {
+  auto maybe_resource = caf::get_as<uri>(cfg, "resource");
+  if (!maybe_resource) {
+    std::cerr << "*** missing mandatory option --resource" << std::endl;
+    return EXIT_FAILURE;
+  }
+  auto resource = std::move(*maybe_resource);
+  auto use_ssl = caf::get_or(cfg, "tls.enable", false);
+  auto ca_file = caf::get_as<std::string>(cfg, "tls.ca-file");
+  if (resource.scheme() != "http" && resource.scheme() != "https") {
+    std::cerr << "*** only HTTP is supported at the moment" << std::endl;
+    return EXIT_FAILURE;
+  }
+  auto method = caf::get_or(cfg, "method", default_method);
+  auto payload = caf::get_or(cfg, "payload", ""sv);
+  auto& auth = resource.authority();
+  auto port = auth.port != 0 ? auth.port : defatul_port(resource.scheme());
+  auto conn
+    = http::with(sys)
+        // Optionally enable TLS.
+        .context(ssl::context::enable(use_ssl)
+                   .and_then(ssl::emplace_client(ssl::tls::v1_2))
+                   .and_then(ssl::load_verify_file_if(ca_file)))
+        // Connect to "$host:$port".
+        .connect(std::get<0>(auth.host), port)
+        // If we don't succeed at first, try up to 5 times with 1s delay.
+        .retry_delay(1s)
+        .max_retry_count(5)
+        // After connecting, send a get request.
+        .request(method, resource.path_query_fragment(), payload);
+  if (!conn) {
+    std::cerr << "*** Failed to initiate connection: "
+              << to_string(conn.error()) << std::endl;
+    return EXIT_FAILURE;
+  }
+  auto testee = sys.spawn([&conn](event_based_actor* self) {
+    conn->first.bind_to(self).then(
+      [](const http::response& r) {
+        std::cout << "HTTP/1.0 " << detail::to_underlying(r.code()) << ' '
+                  << phrase(r.code()) << std::endl;
+        // hdr.version()  hdr.status_text()
+        for (const auto& [key, value] : r.header_fields())
+          if (key != "Date")
+            std::cout << key << ':' << value << std::endl;
+        std::cout << std::string_view{reinterpret_cast<const char*>(
+                                        r.body().data()),
+                                      r.body().size()}
+                  << std::endl;
+      },
+      [](const error& err) {
+        std::cerr << "*** Error received: " << to_string(err) << std::endl;
+      });
+  });
+  sys.await_all_actors_done();
```

Comment:
Why did you put this here? CAF will wait for `testee` anyways, no?

Reply:
That was my thought as well, but when I remove this line, I get a segfault with the location pointing to `future::then`. I will examine this to get more detail, maybe it was caused by something else.

Reply:
Taking the capture by copy solved this.